app/src/main/java/com/m/services/auth/LoginController.java

```diff
@@ -111,7 +101,6 @@ private void receivedTokenResponse(
             @Nullable AuthorizationException authException) {
         mAuthState.update(tokenResponse, authException);
 
-        System.out.println("Token expires in : " + mAuthState.getAccessTokenExpirationTime());
-        startActivity(new Intent(this, MainActivity.class));
+        listener.authComplete();
```

Comment:
By the look of the code, it looks like if there is an error during login, user won't get notifications about it. A small improvement we can do is to add a error handler to the listener, and show errors to the user if there is any.

Reply:
I have done some work here but not sure if its what you had meant.

Reply:
It's half way there. What I mean is we should do the same as how errors are handled in the TaskActivity: when there is an error, pass it back to the activity and show a notification about the error.